src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -680,15 +680,28 @@ public function getAutowired()
      *
      * Allowed values:
      *   - true: constructor autowiring
-     *   - array('__construct', 'setFoo', 'setBar'): autowire whitelisted methods only
-     *   - '*': autowire constructor and every setters
+     *   - array('__construct', 'set*', 'setBar'): autowire whitelisted methods only
      *
-     * @param bool|string|string[] $autowired
+     * @param string[]|bool $autowired
      *
      * @return Definition The current instance
      */
     public function setAutowired($autowired)
     {
+        if (false === $autowired) {
+            $this->autowired = array();
+
+            return $this;
+        }
+
+        if (true === $autowired) {
```

Comment:
I think this case should be more predictable and never take the previous value into account: 
`$autowired = array('__construct')`